back/engines/commercial/multi_tenancy/spec/tasks/insert_option_key_values_in_user_custom_field_values_spec.rb

```diff
@@ -0,0 +1,117 @@
+
+  it 'does not overwite an existing value when the given value is nil' do
+    expect(csv_data[4]['value']).to be_nil
+    expect(User.find(csv_data[4]['id']).custom_field_values[custom_field.key]).to eq(custom_field_option1.key)
+
+    task.invoke(csv, 'example.org', custom_field.id, 'execute')
+
+    expect(User.find(csv_data[4]['id']).custom_field_values[custom_field.key]).to eq(custom_field_option1.key)
+  end
+
+  it 'does not overwite an existing value when the given value is valid' do
+    expect(csv_data[5]['value']).to eq('1234AB')
+    expect(User.find(csv_data[5]['id']).custom_field_values[custom_field.key]).to eq(custom_field_option2.key)
+
+    task.invoke(csv, 'example.org', custom_field.id, 'execute')
+
+    expect(User.find(csv_data[5]['id']).custom_field_values[custom_field.key]).to eq(custom_field_option2.key)
+  end
+
+  it "does nothing if the 4th argument is not 'execute'" do
+    task.invoke(csv, 'example.org', custom_field.id, 'dry_run')
+
+    expect(User.find(csv_data[0]['id']).custom_field_values).to eq({ 'unrelated_key' => 'some_value' })
+    expect(User.find(csv_data[1]['id']).custom_field_values).to eq({})
+    expect(User.find(csv_data[2]['id']).custom_field_values).to eq({})
+    expect(User.find(csv_data[3]['id']).custom_field_values).to eq({})
+    expect(User.find(csv_data[4]['id']).custom_field_values)
+      .to eq({ custom_field.key.to_s => custom_field_option1.key })
+    expect(User.find(csv_data[5]['id']).custom_field_values)
+      .to eq({ custom_field.key.to_s => custom_field_option2.key })
+  end
+
+  it 'does nothing if the 4th argument is not given' do
+    task.invoke(csv, 'example.org', custom_field.id)
+
+    expect(User.find(csv_data[0]['id']).custom_field_values).to eq({ 'unrelated_key' => 'some_value' })
+    expect(User.find(csv_data[1]['id']).custom_field_values).to eq({})
+    expect(User.find(csv_data[2]['id']).custom_field_values).to eq({})
+    expect(User.find(csv_data[3]['id']).custom_field_values).to eq({})
+    expect(User.find(csv_data[4]['id']).custom_field_values)
+      .to eq({ custom_field.key.to_s => custom_field_option1.key })
+    expect(User.find(csv_data[5]['id']).custom_field_values)
+      .to eq({ custom_field.key.to_s => custom_field_option2.key })
+  end
+
+  it 'does nothing if the custom_field does not exist' do
+    task.invoke(csv, 'example.org', 'not_a_custom_field_id', 'execute')
+
+    expect(User.find(csv_data[0]['id']).custom_field_values).to eq({ 'unrelated_key' => 'some_value' })
+    expect(User.find(csv_data[1]['id']).custom_field_values).to eq({})
+    expect(User.find(csv_data[2]['id']).custom_field_values).to eq({})
+    expect(User.find(csv_data[3]['id']).custom_field_values).to eq({})
+    expect(User.find(csv_data[4]['id']).custom_field_values)
+      .to eq({ custom_field.key.to_s => custom_field_option1.key })
+    expect(User.find(csv_data[5]['id']).custom_field_values)
+      .to eq({ custom_field.key.to_s => custom_field_option2.key })
+  end
```

Comment:
I spent quite some time trying to find a way to remove the obvious duplication in these tests, but was unable to get any approach to work. I tried shared_examples and parameterized tests.

Probably with more time, I can manage it, but I'm not convinced it's worth the time investment.